packages/core/src/sync-store/index.ts

```diff
@@ -166,329 +161,75 @@ export const createSyncStore = ({
-                  .$call((qb) => topicSQL(qb, fragment));
-            }
-          }
-          break;
-
-        case "block":
-          {
-            fragments = [buildBlockFilterFragment(filter)];
-            table = "blockFilter";
-            idCol = "blockFilterId";
-            fragmentSelect = (fragment, qb) =>
-              qb.where("blockFilterId", "=", fragment.id);
-          }
-          break;
 
-        case "callTrace":
-          {
-            if (isAddressFactory(filter.toAddress)) {
-              fragments = buildTraceFilterFragments(filter);
-              table = "factoryTraceFilter";
-              idCol = "factoryId";
-              fragmentSelect = (fragment: TraceFilterFragment<Factory>, qb) =>
-                qb
-                  .where("address", "=", fragment.address)
-                  .where("eventSelector", "=", fragment.eventSelector)
-                  .where(
-                    "childAddressLocation",
-                    "=",
-                    fragment.childAddressLocation,
-                  )
-                  .where((eb) =>
-                    eb.or([
-                      eb("fromAddress", "is", null),
-                      eb("fromAddress", "=", fragment.fromAddress),
-                    ]),
-                  );
-            } else {
-              fragments = buildTraceFilterFragments(filter);
-              table = "traceFilter";
-              idCol = "traceFilterId";
-              fragmentSelect = (fragment: TraceFilterFragment<undefined>, qb) =>
-                qb
-                  .where((eb) =>
-                    eb.or([
-                      eb("fromAddress", "is", null),
-                      eb("fromAddress", "=", fragment.fromAddress),
-                    ]),
-                  )
-                  .where((eb) =>
-                    eb.or([
-                      eb("toAddress", "is", null),
-                      eb("toAddress", "=", fragment.toAddress),
-                    ]),
-                  );
-            }
-          }
-          break;
+      await db
+        .insertInto("interval")
+        .values(values)
+        .onConflict((oc) =>
+          oc.column("fragment_id").doUpdateSet({
+            blocks: ksql`interval.blocks + excluded.blocks`,
+          }),
+        )
+        .execute();
+    });
+  },
+  getIntervals: async ({ filters }) =>
+    db.wrap({ method: "getIntervals" }, async () => {
+      let query:
+        | SelectQueryBuilder<
+            PonderSyncSchema,
+            "interval",
+            { blocks: string; filter: string }
+          >
+        | undefined;
+
+      for (let i = 0; i < filters.length; i++) {
+        const filter = filters[i]!;
+        const fragments = getFragmentIds(filter);
+        for (const fragment of fragments) {
+          const _query = db
+            .selectFrom("interval")
+            .where("fragment_id", "in", fragment.adjacent)
+            .select(["blocks", sql<string>`${i}`.as("filter")]);
```

Comment:
I think there's a small bug here where the adjacent range information is getting lost. Ideally we could union the multiranges for all the adjacent fragments that make up this fragment right here in this select statement. Maybe the `SUM` operator works?

https://www.postgresql.org/docs/current/functions-range.html#MULTIRANGE-OPERATORS-TABLE